examples/models/models_first_person_maze.cpp

```diff
@@ -0,0 +1,137 @@
+
+        int playerCellX = static_cast<int>(playerPos.x - mapPosition.x + 0.5f);
+        int playerCellY = static_cast<int>(playerPos.y - mapPosition.z + 0.5f);
+
+        // Out-of-limits security check
+        if (playerCellX < 0) playerCellX = 0;
+        else if (playerCellX >= cubicmap.width) playerCellX = cubicmap.width - 1;
+
+        if (playerCellY < 0) playerCellY = 0;
+        else if (playerCellY >= cubicmap.height) playerCellY = cubicmap.height - 1;
+
+        // Check map collisions using image data and player position
+        // TODO: Improvement: Just check player surrounding cells for collision
+        for (int y = 0; y < cubicmap.height; y++)
+        {
+            for (int x = 0; x < cubicmap.width; x++)
+            {
+                if ((mapPixels[y*cubicmap.width + x].r == 255) &&       // Collision: white pixel, only check R channel
+                    (playerPos.CheckCollisionCircleRec(playerRadius,
+                    (Rectangle){ mapPosition.x - 0.5f + x*1.0f, mapPosition.z - 0.5f + y*1.0f, 1.0f, 1.0f })))
+                {
+                    // Collision detected, reset camera position
+                    camera.position = oldCamPos;
+                }
+            }
+        }
+        //----------------------------------------------------------------------------------
+
+        // Draw
+        //----------------------------------------------------------------------------------
+        BeginDrawing();
+
+            ClearBackground(RAYWHITE);
+
+            camera.BeginMode3D();
+
+                model.Draw(mapPosition);               // Draw maze map
+                // DrawCubeV(playerPosition, (Vector3){ 0.2f, 0.4f, 0.2f }, RED);  // Draw player
+
+            camera.EndMode3D();
+
+            cubicmap.Draw((Vector2){ GetScreenWidth() - cubicmap.width*4 - 20, 20 }, 0.0f, 4.0f, WHITE);
+            DrawRectangleLines(GetScreenWidth() - cubicmap.width*4 - 20, 20, cubicmap.width*4, cubicmap.height*4, GREEN);
+
+            // Draw player position radar
+            DrawRectangle(GetScreenWidth() - cubicmap.width*4 - 20 + playerCellX*4, 20 + playerCellY*4, 4, 4, RED);
+
+            DrawFPS(10, 10);
+
+        EndDrawing();
+        //----------------------------------------------------------------------------------
+    }
+
+    // De-Initialization
+    //--------------------------------------------------------------------------------------
+    free(mapPixels);            // Unload color array
+
+    /**
+     * TODO: Fix double freeing of Mesh, possibly from UnloadModel() unloading meshes in Model.hpp?
+     *
+     * INFO: TEXTURE: [ID 4] Unloaded texture data from VRAM (GPU)
+     * INFO: VAO: [ID 2] Unloaded vertex data from VRAM (GPU)
+     * INFO: MODEL: Unloaded model (and meshes) from RAM and VRAM
+     * double free or corruption (!prev)
+    */
+    mesh.vboId = NULL; // This shouldn't be needed, but currently is.
```

Comment:
@stephenaa With this change, you can tell the Mesh not to unload since it's being unloaded by the Model, but this is a gross hack. :thinking: